08-generic_cache/02_mutex/cache_internal_test.go

```diff
@@ -0,0 +1,45 @@
+package cache
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCacheRead(t *testing.T) {
```

Comment:
I don't see a reason to have internal tests here, it would be better to instead have more public API tests. Internal tests are usually more fragile when making modifications to the implementation.